### ISSUE TITLE
Unable to rename database if a table exists that references a sequence

### ISSUE THREAD

**Issue by florence-crl** [cockroachdb/cockroach#45411]
**Describe the problem**
You can not rename a database if a table references a sequence. The error message is not accurate.

**To Reproduce**
with `cockroach sql`

```
defaultdb> CREATE DATABASE db1;
defaultdb> USE db1;
db1> CREATE SEQUENCE seq1 MINVALUE 1 MAXVALUE 9223372036854775807 INCREMENT 1 START 1;
db1> CREATE TABLE tab1 (
  id INT8 NOT NULL DEFAULT nextval('seq1':::STRING),
  name STRING NULL,
  CONSTRAINT "primary" PRIMARY KEY (id ASC),
  FAMILY "primary" (id, name)
);
root@localhost:26257/db1> use defaultdb;
root@localhost:26257/defaultdb> alter database db1 rename to db2;
pq: cannot rename database because view "tab1" depends on table "seq1"
HINT: you can drop tab1 instead.
```

In error message , `view "tab1"` is actually a table and `table "seq1"` is actually a sequence.

**Expected behavior**
I expect the database to be renamed without error.

**Additional data / screenshots**
The error is generated by `cockroach/pkg/sql/rename_database.go` which has the comment below so it is intended for views. 

```
    // Check if any views depend on tables in the database. Because our views
    // are currently just stored as strings, they explicitly specify the database
    // name. Rather than trying to rewrite them with the changed DB name, we
    // simply disallow such renames for now
```

**Environment:**
 - CockroachDB version:  19.2.4
 - Server OS:  Linux
 - Client app: `cockroach sql`

**Additional context**
What was the impact?
The customer is not able to rename their database without dropping the table that uses a sequence.

I am not sure if this issue is the same as https://github.com/cockroachdb/cockroach/issues/34416

**Comment by RoachietheSupportRoach** [comment]
Zendesk ticket #[4797](https://cockroachdb.zendesk.com/agent/tickets/4797) has been linked to this issue.

**Comment by otan** [comment]
This is kind of related to #34416 for the reason described there - we can reference separate databases in cockroach (whereas postgres can only reference cross-schema) -- as an example, if we have `nextval('different_db.public.table_name')` and we rename the database `different_db`, the `nextval` value will be incorrect as it would reference nothing (we have a three level schema system: https://www.cockroachlabs.com/docs/stable/show-schemas.html). postgres allows this as we cannot reference a separate database in `nextval`. 

The error message, on the other hand, I will fix, then deferring to #34416.

**Comment by menih** [comment]
What would be the workaround for this than? All out tables will have a sequence associated with them. Are we to drop them all, just to be able to rename db?

In case you are wondering what the use case - we need to backup / restore the whole DB, so when restoring, we create a new DB, restore to it, and than rename current DB (so we don't loose it), and than rename the "restore db" to the current.

If I am going about it the wrong way, would be happy to get any suggestions on alternative. But otherwise, we are now blocked with out evaluation of CRDB.

**Comment by otan** [comment]
Hey @menih, I can see the pain point for sure. At the moment, yes this is definitely a problem for your rename case, I can't think of a workaround at the moment (let me ask around).

Just out of curiousity, does that `nextval` have a database name inside it (e.g. `db_name.sequence_name`), or is it just the sequence name (assuming it's in the same database, e.g. `nextval('sequence_name')`. In theory, we could allow renames of databases with sequences which do not reference other databases provided the `nextval` value does not reference another database name.

**Comment by menih** [comment]
Hi Oliver,

Thanks for quick reply.

Interestingly, the sequences are all created automatically by CRDB. We use
the "SET experimental_serial_normalization=sql_sequence;" when I create
tables as we need to ensure ID column is a running number.

And it seems that neither the sequence, nor the table which uses it, has
the DB name incorporated in it. So yes, this should work for us once fixed.

Can you give me an idea how complicated is to fix?

Thanks,
Meni

**Comment by otan** [comment]
Ah sorry I edited my question and judging by the email reply you might have missed it. I was wondering why you wanted strict sequencing by turning that on instead of using something like unique_rowid (see considerations in https://www.cockroachlabs.com/docs/stable/create-sequence.html).

It's a little finicky to fix - we will definitely review it in our roadmap.

**Comment by menih** [comment]
Some of our IDs are user facing, some are limited in size, and for
troubleshooting/usability, it is much easier to have a running number. Yes,
I am aware of the performance impact, especially in a cluster setting...
but willing to pay for it.

Meni

**Comment by otan** [comment]
@menih this will be resolved for your use case in 20.1
